### PR TITLE
test: address Jest test timeout issues

### PR DESCRIPTION
**Pull request by PenghaiZhang**
##### Checklist

- [x  the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [x] tests are included
- [x] screenshots are included showing significant UI changes
- [ ] documentation is changed or added

##### Description of change

The Jest test case of LTI Platform creation was too big and GHA builds failed on it due to timeout very frequently. As a result, I split the test case into 5 smaller cases and reduced the timeout for each to 20 seconds. The overall time taken by this test suite increases, but it should be a lot more stable...so I think it's worth.

image

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by PenghaiZhang** [comment]
The test was happy in the first run. Now try again.

**Comment by PenghaiZhang** [comment]
I noticed another test failed on timeout.

> FAIL __tests__/tsrc/components/securityentitydialog/SelectGroupDialog.test.tsx (10.403 s)
  ● SelectGroupDialog › Should be able to cancel the action
    thrown: "Exceeded timeout of 5000 ms for a test.

**Comment by PenghaiZhang** [comment]
The LTI Platform test was happy in the second run. Now try again.

**Comment by PenghaiZhang** [comment]
The LTI Platform test was happy in the third run.

But below tests also require some refactoring.

* SelectGroupDialog.test.tsx
* SelectRoleDialog.test.tsx 
* ACLExpressionBuilder.test.tsx

**Comment by PenghaiZhang** [comment]
For the two dialog ones, I can only think of extending the timeout to 15s for them. And for the ACL expression builder one, after looking into the test I am not sure whether we skip some UI operations in the test. Do you have any idea @edalex-yinzi  ?

**Comment by edalex-yinzi** [comment]
> For the two dialog ones, I can only think of extending the timeout to 15s for them. And for the ACL expression builder one, after looking into the test I am not sure whether we skip some UI operations in the test. Do you have any idea @edalex-yinzi ?

I will have a look，so which operations are missing？

**Comment by PenghaiZhang** [comment]
> > For the two dialog ones, I can only think of extending the timeout to 15s for them. And for the ACL expression builder one, after looking into the test I am not sure whether we skip some UI operations in the test. Do you have any idea @edalex-yinzi ?
> 
> I will have a look，so which operations are missing？

Nothing is missing I guess, but I was wondering if we could skip any operation to make the test faster .

**Comment by PenghaiZhang** [comment]
**ACLExpressionBuilder.test.tsx** now runs much faster - over 50% less time saved.

Before:  
image


After:

image

**Comment by PenghaiZhang** [comment]
Here is a summary for all the changes made in this PR.

* Split LTI platform creation test case into multiple cases;
* Extend the timeout of two cases in `SelectGroupDialog` and `SelectRoleDialog` respectively;
* Optimise the test cases of `ACLExpressionBuilder` by using `Select all` button instead of adding entities one by one

**Comment by PenghaiZhang** [comment]
All three tests passed in the first run.

**Comment by PenghaiZhang** [comment]
All three tests passed in the second run.

**Comment by PenghaiZhang** [comment]
All three tests passed in the third run.

**Comment by edalex-yinzi** [comment]
image

**Comment by edalex-yinzi** [comment]
Start the fourth time.

**Review by edalex-yinzi — COMMENTED** [review]
Thanks Penghai, one query for you.

File: react-front-end/__tests__/tsrc/components/aclexpressionbuilder/ACLExpressionBuilderTestHelper.tsx [hunk 1]
```diff
@@ -60,16 +64,15 @@ export const renderACLExpressionBuilder = (
...
+ * Select all the entities shown in the `EntitySearch` (User/Group/RoleSearch) result list by
+ * clicking the `Select all` button and then `ok` button.
  */
-export const selectAndConfirm = async (
-  container: HTMLElement,
-  selectNames: string[],
-) => {
-  // wait for the results, and then click all entities
-  await selectEntitiesInOneClickMode(container, selectNames);
-  // click ok button
```
[the remark this answers is not in the document]
Comment (PenghaiZhang) [reply react-front-end/__tests__/tsrc/components/aclexpressionbuilder/ACLExpressionBuilderTestHelper.tsx:72]: Say you have three entities to select. The previous way is to select them one by one, which means 3 UI interactions.  The new way is to use the `Select all` button, and it just needs 1 click. 

If the test targets some very specific data, you may want to select them individually.
But this is not case in these tests as far as I can tell.

**Pull request merged**